qiime2/core/util.py

```diff
@@ -406,11 +434,11 @@ def ref_constructor(loader, node):
     def cite_constructor(loader, node):
         return node.value
 
-    def metadata_constructor(loader, node):
-        # Use the md5sum of the metadata as its identifier, so we can tell
+    def metadata_constructor(loader, node, checksum_type):
```

Comment:
We should have seen a failing test for this. The metadata_constructor is used a few lines under this and doesn't have any way to take `checksum_type` as an argument. Also, the metadata entries in provenance are still fixed to md5, so we can keep it md5 here as well.

Reply:
i'll add the default arg as md5 here - re: test failures, i'd like to chat about this for a few mins about this so i can add anything that's been previously missing

Reply:
evan and i chatted offline about this - there was a very unusual chain of confusion that's been resolved w/an added comment on how this constructor works wrt provenance's constructor. closing the case on this.